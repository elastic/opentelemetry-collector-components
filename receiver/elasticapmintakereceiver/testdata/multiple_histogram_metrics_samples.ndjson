{"metadata": {"service": {"name": "my-service", "version": "1.0.0", "agent": {"version": "3.14.0", "name": "elastic-node"}}}}
{"metricset": {"samples": {"valid_histogram_metric_1": {"type": "histogram", "unit": "ms", "counts": [1, 3, 5], "values": [10, 20, 30]}}, "timestamp": 1757462026684385}}
{"metricset": {"samples": {"valid_histogram_metric_2": {"type": "histogram", "unit": "ms", "counts": [2, 4, 6, 8, 10], "values": [5, 10, 15, 25, 50]}}, "timestamp": 1757462026684385}}
{"metricset": {"samples": {"value_count_mismatch_histogram_metric": {"type": "histogram", "unit": "ms", "counts": [1, 2, 3], "values": [10, 20]}}, "timestamp": 1757462026684385}}
