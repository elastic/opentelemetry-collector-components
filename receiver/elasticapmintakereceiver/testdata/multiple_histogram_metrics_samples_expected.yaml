resourceMetrics:
  - resource:
      attributes:
        - key: agent.name
          value:
            stringValue: elastic-node
        - key: agent.version
          value:
            stringValue: 3.14.0
        - key: service.name
          value:
            stringValue: my-service
        - key: service.version
          value:
            stringValue: 1.0.0
        - key: telemetry.sdk.name
          value:
            stringValue: ElasticAPM
    scopeMetrics:
      - metrics:
          - histogram:
              dataPoints:
                - attributes:
                    - key: event.outcome
                      value:
                        stringValue: unknown
                    - key: processor.event
                      value:
                        stringValue: metric
                    - key: timestamp.us
                      value:
                        intValue: "1757462026684385"
                  bucketCounts:
                    - "1"
                    - "3"
                    - "5"
                  count: "9"
                  explicitBounds:
                    - 10
                    - 20
                  sum: 220
                  timeUnixNano: "1757462026684385000"
            name: valid_histogram_metric_1
            unit: ms
        scope: {}
  - resource:
      attributes:
        - key: agent.name
          value:
            stringValue: elastic-node
        - key: agent.version
          value:
            stringValue: 3.14.0
        - key: service.name
          value:
            stringValue: my-service
        - key: service.version
          value:
            stringValue: 1.0.0
        - key: telemetry.sdk.name
          value:
            stringValue: ElasticAPM
    scopeMetrics:
      - metrics:
          - histogram:
              dataPoints:
                - attributes:
                    - key: event.outcome
                      value:
                        stringValue: unknown
                    - key: processor.event
                      value:
                        stringValue: metric
                    - key: timestamp.us
                      value:
                        intValue: "1757462026684385"
                  bucketCounts:
                    - "2"
                    - "4"
                    - "6"
                    - "8"
                    - "10"
                  count: "30"
                  explicitBounds:
                    - 5
                    - 10
                    - 15
                    - 25
                  sum: 840
                  timeUnixNano: "1757462026684385000"
            name: valid_histogram_metric_2
            unit: ms
        scope: {}
  - resource:
      attributes:
        - key: agent.name
          value:
            stringValue: elastic-node
        - key: agent.version
          value:
            stringValue: 3.14.0
        - key: service.name
          value:
            stringValue: my-service
        - key: service.version
          value:
            stringValue: 1.0.0
        - key: telemetry.sdk.name
          value:
            stringValue: ElasticAPM
    scopeMetrics:
      - metrics:
          - histogram:
              dataPoints:
                - attributes:
                    - key: event.outcome
                      value:
                        stringValue: unknown
                    - key: processor.event
                      value:
                        stringValue: metric
                    - key: timestamp.us
                      value:
                        intValue: "1757462026684385"
                  timeUnixNano: "1757462026684385000"
            name: value_count_mismatch_histogram_metric
            unit: ms
        scope: {}
